include/s3select_oper.h

```diff
@@ -1464,33 +1475,44 @@ class logical_operand : public base_statement
     value a = l->eval();
     if (_oplog == oplog_t::AND)
     {
-      if (a.i64() == false)
-      {
+      if (!a.is_null() && a.i64() == false) {
         res = false ^ negation_result;
         return var_value = res;
       } 
       value b = r->eval();
-      if (a.is_null() || b.is_null()) {
-        var_value.setnull();
+      if((!a.is_null() && a.i64() == false) || (!b.is_null() && b.i64() == false)) {
+        res = false ^ negation_result;
+        return var_value = res;
       } else {
-        res =  (a.i64() && b.i64()) ^ negation_result ;
-      }
+        if (a.is_null() || b.is_null()) {
+          var_value.setnull();
+          return var_value;
+        } else {
+          res =  (a.i64() && b.i64()) ^ negation_result ;
+          return var_value =res;
+        }
+      }   
     }
     else
     {
-      if (a.i64() == true)
-      {
+      if (a.i64() == true) {
         res = true ^ negation_result;
         return var_value = res;
-      }
+      } 
       value b = r->eval();
-      if (a.is_null() || b.is_null()) {
-        var_value.setnull();
+      if(a.i64() ==  true || b.i64() == true) {
+        res = true ^ negation_result;
+        return var_value = res;
       } else {
-        res =  (a.i64() || b.i64()) ^ negation_result;
+        if (a.is_null() || b.is_null()) {
+          var_value.setnull();
+          return var_value;
+        } else {
+          res =  (a.i64() || b.i64()) ^ negation_result ;
```

Comment:
at this point , a and b , are not null nor true.
no need for /(a.i64() || b.i64())/

Reply:
ok, will just return a false there

Reply:
not to forget / ^ negation_result/